egs/wsj/s5/utils/validate_data_dir.sh

```diff
@@ -179,7 +179,7 @@ if [ -f $data/wav.scp ]; then
     check_sorted_and_uniq $data/segments
     # We have a segments file -> interpret wav file as "recording-ids" not utterance-ids.
     ! cat $data/segments | \
-      awk '{if (NF != 4 || $4 <= $3) { print "Bad line in segments file", $0; exit(1); }}' && \
+      awk '{if (NF != 4 || ( $4 > 0 && $4 <= $3 ) ) { print "Bad line in segments file", $0; exit(1); }}' && \
```

Comment:
Just a suggestion: awkly use of awk gets rid of the if and a couple of pairs of braces.
```suggestion
      awk 'NF != 4 || ($4 > 0 && $4 <= $3) {
             print "Bad line", NR, "in segments file:", $0; exit(1); }' && \
```